src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java

```diff
@@ -262,18 +263,7 @@ private void readLoop(final SeekableByteChannel reader) throws IOException, Inte
                         // Return to the file loop
                         return;
                     }
-                    try {
-                        lastChecked = Optional.of(Files.getLastModifiedTime(_file).toMillis());
-                    } catch (final NoSuchFileException t) {
-                        rotate(
-                                Optional.of(reader),
-                                String.format(
-                                        "File rotation detected based last modified time access failure; file=%s",
-                                        _file));
-
-                        // Return to the file loop
-                        return;
-                    }
+                    lastChecked = Optional.of(attributes.getLastModifiedTime());
```

Comment:
How are you guaranteed that this won't fail if the file does not exist? Is the attributes one the new way of doing this?